internal/pkg/app/handlers_v1.go

```diff
@@ -224,14 +225,17 @@ func (a *APIManager) sendEventHandler(w http.ResponseWriter, r *http.Request) {
 		resp.Respond(ctx, w, doYaml(r))
 		return
 	}
-	err = a.routingTableMgr.RouteEvent(ctx, *tid, routeId, payload)
+	traceId, err := a.routingTableMgr.RouteEvent(ctx, *tid, routeId, payload)
 	if err != nil {
 		log.Ctx(ctx).Error().Str("op", "sendEventHandler").Msg(err.Error())
 		resp := ErrorResponse(convertToApiError(ctx, err))
 		resp.Respond(ctx, w, doYaml(r))
 		return
 	}
-	resp := ItemResponse(routeId)
+	item := make(map[string]string)
+	item["routeId"] = routeId
+	item["tx.traceId"] = traceId
+	resp := ItemResponse(item)
 	resp.Respond(ctx, w, doYaml(r))
```

Comment:
Also, you *might* get trace.id for free if you respond with event's context (but then u will need to return the event instance from the RouteEvent method

Reply:
yeah i was wondering if that would always work - depending on timing the context may already be canceled, no?